src/osd/PG.cc

```diff
@@ -4547,39 +4547,47 @@ void PG::share_pg_info()
 
 void PG::append_log_entries_update_missing(
   const list<pg_log_entry_t> &entries,
+  bool handle_missing,
   ObjectStore::Transaction &t)
 {
   assert(!entries.empty());
   assert(entries.begin()->version > info.last_update);
 
-  PGLogEntryHandler rollbacker;
-  pg_log.append_new_log_entries(
-    info.last_backfill,
-    info.last_backfill_bitwise,
-    entries,
-    &rollbacker);
-  rollbacker.apply(this, &t);
-  info.last_update = pg_log.get_head();
+  if (handle_missing) {
+    PGLogEntryHandler rollbacker;
```

Comment:
It seems like it would be easier to just update append_new_log_entries to do the right thing.  Also, append_new_log_entries could indicate whether stats should be invalidated based on the log entries.  That would avoid needing to update the message.